### PR TITLE
added circleci as gh action YAML

### LINKED ISSUE #1280: Move to CI for tests to GH actions
Let's get consistent and move [Circle CI](https://github.com/prometheus/client_golang/tree/main/.circleci) tests to GH Actions which are generally less problematic (recently we had issue with hooks [denied](https://github.com/prometheus/client_golang/pull/1278) from contributors?). [We can copy and adjust Prometheus yaml](https://github.com/prometheus/prometheus/blob/main/.github/workflows/ci.yml)

### PR DESCRIPTION
**Pull request by krishnaduttPanchagnula**
Signed-off-by: Krishna Dutt Panchagnula <krishnadutt123@gmail.com>
closes #1280

### CODE REVIEW

**Comment by bwplotka** [comment]
Amazing, thanks for this! Just approved now, sorry for lag, let's get this done this week 💪🏽

**Comment by bwplotka** [comment]
Also DCO is required (:

File: .github/workflows/Circle-ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go_version: ["1.18", "1.19", "1.20"]
+        run_lint: [true, true, true]
```
[the remark this answers is not in the document]
Comment (krishnaduttPanchagnula) [reply .github/workflows/Circle-ci.yml]: @bwplotka you are right, i was thinking to use it as pairs, but thats not how the matrix would work, right?

 I have updated the code with a simpler version. Please review and let me know if it works?

**Comment by krishnaduttPanchagnula** [comment]
> this yaml, on your own or from Prometheus

No, i created that based on my research.

> matrix configuration. I would just remove those bool flags.

@bwplotka  i resolved it. let me know if that works ?

**Comment by bwplotka** [comment]
Can we rebase / squash to just one commit? Somehow this change is adding more things than it should (:

File: .github/workflows/Circle-ci.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          key: v1-go${{ matrix.go_version }}
+
+      - name: Run tests and check license
+        run: make check_license test
+        env:
+          CI: true
+
+      - name: Run lint
+        if: ${{ matrix.go_version == '1.18' || matrix.go_version == '1.19' }}
+        run: make lint
```
Comment (bwplotka) [inline .github/workflows/Circle-ci.yml:42]: Any particular reason to exclude linting?

Comment (krishnaduttPanchagnula) [reply .github/workflows/Circle-ci.yml:42]: My bad. Changed the code.

**Comment by krishnaduttPanchagnula** [comment]
I tried rebasing and reverting those chanes in my forked repo, but they are not reflecting here, Is it okey if i raise another PR with same name @bwplotka ?

**Comment by krishnaduttPanchagnula** [comment]
@bwplotka  resolved the issue. Please take a look and comment if any other changes are to be done

**Pull request merged**